extractor/src/main/java/org/schabi/newpipe/extractor/services/peertube/extractors/PeertubeStreamExtractor.java

```diff
@@ -236,10 +236,16 @@ public StreamInfoItemsCollector getRelatedStreams() throws IOException, Extracti
         return collector;
     }
 
+    @Nonnull
     @Override
     public List<String> getTags() {
         try {
-            return (List) JsonUtils.getArray(json, "tags");
+            JsonArray tagsArray = json.getArray("tags");
+            List<String> tagsList = new ArrayList<>();
```

Comment:
1. I'm interested in why the `JsonUtils` is not sufficient here. Is it for performance reasons or the the unchecked cast?
2. This section to transform the `JsonArray` to a `ArrayList` is written once for youtube, peertube and mediaccc. This could be extracted into the `JsonUtils`

Reply:
@XiangRongLin is right that this code is stupid. `return json.getArray("tags");` would be sufficent, as it returns a `JsonArray`, which extends `ArrayList`.

Reply:
no, because then you return a JsonArray, i.e. ArrayList**<Object>**, and we want a List**<String>**. It must have Strings only, and I wasn't sure if the cast would be everytime safe. I'm going to revert then.